### DISCUSSION TITLE
详解 Kubernetes ReplicaSet 的实现原理 · /kubernetes-replicaset

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#547] [Announcements]
https://draveness.me/kubernetes-replicaset

**Comment by jinrunsen** [comment]
大佬，图显示不出来了

**Comment by X-shuffle** [comment]
大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。

**Comment by draveness** [comment]
> 大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。

你可以直接 edit 这个 Pod 🙂

**Comment by X-shuffle** [comment]
> @draveness
> > 大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。
> 
> 你可以直接 edit 这个 Pod 🙂

有个问题就是，我想实现滚动的一个效果。因为我不确定本次升级是否能够成功。要保证时时存在一个可用的版本。换句话说，可不可以让这个pod被另外一个deployment控制，用来实现滚动更新的效果

**Comment by draveness** [comment]
> > @draveness
> > > 大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。
> > 
> > 
> > 你可以直接 edit 这个 Pod 🙂
> 
> 有个问题就是，我想实现滚动的一个效果。因为我不确定本次升级是否能够成功。要保证时时存在一个可用的版本。换句话说，可不可以让这个pod被另外一个deployment控制，用来实现滚动更新的效果

有专门做这个的 Operator，类似 https://github.com/etiennecoutaud/kanary

**Reply by dduo518** [reply]
这个现在是不是可以使用滚动更新策略，保留原来的pod，到多久时间之后再全部更新？

**Comment by X-shuffle** [comment]
> > > @draveness
> > > > 大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。
> > > 
> > > 
> > > 你可以直接 edit 这个 Pod 🙂
> > 
> > 
> > 有个问题就是，我想实现滚动的一个效果。因为我不确定本次升级是否能够成功。要保证时时存在一个可用的版本。换句话说，可不可以让这个pod被另外一个deployment控制，用来实现滚动更新的效果
> 
> 有专门做这个的 Operator，类似 https://github.com/etiennecoutaud/kanary

其实我想知道实现原理。想自己通过java调k8s api写一个后台来实现我的想法

**Comment by draveness** [comment]
> > > > @draveness
> > > > > 大佬，我想问下，有没有一中办法，只升级一个deployment下的一个pod，其他pod不变的。
> > > > 
> > > > 
> > > > 你可以直接 edit 这个 Pod 🙂
> > > 
> > > 
> > > 有个问题就是，我想实现滚动的一个效果。因为我不确定本次升级是否能够成功。要保证时时存在一个可用的版本。换句话说，可不可以让这个pod被另外一个deployment控制，用来实现滚动更新的效果
> > 
> > 
> > 有专门做这个的 Operator，类似 https://github.com/etiennecoutaud/kanary
> 
> 其实我想知道实现原理。想自己通过java调k8s api写一个后台来实现我的想法

实现原理就是 CRD 和 Operator，找个 Kubernetes 的 Java Client，但是这个一般 Go 比较好写

**Comment by zhouwentong1993** [comment]
流程图没有渲染出来，其它文章也是如此。
